File: src/test/java/fi/metatavu/metaform/test/functional/tests/AnswererTestsIT.java
package fi.metatavu.metaform.test.functional.tests;

import fi.metatavu.metaform.api.client.models.Metaform;
import fi.metatavu.metaform.api.client.models.Reply;
import fi.metatavu.metaform.server.rest.ReplyMode;
import fi.metatavu.metaform.test.functional.AbstractIntegrationTest;
import fi.metatavu.metaform.test.functional.builder.TestBuilder;
import fi.metatavu.metaform.test.functional.builder.resources.KeycloakResource;
import fi.metatavu.metaform.test.functional.builder.resources.MysqlResource;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.quarkus.test.junit.TestProfile;
import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

/**
 * Tests for testing answerer user form functionality
 */
@QuarkusTest
@QuarkusTestResource.List(value = {
  @QuarkusTestResource(MysqlResource.class),
  @QuarkusTestResource(KeycloakResource.class)
})
@TestProfile(GeneralTestProfile.class)
public class AnswererTestsIT extends AbstractIntegrationTest {

  @Test
  public void testAnswererCreateForm() throws Exception {
    try (TestBuilder testBuilder = new TestBuilder()) {
      Metaform payload = new Metaform();
      testBuilder.answerer1.metaforms().assertCreateFailStatus(403, payload);
    }
  }

  @Test
  public void testAnswererFindForm() throws Exception {
    try (TestBuilder testBuilder = new TestBuilder()) {
      Metaform metaform = testBuilder.metaformAdmin.metaforms().createFromJsonFile("simple");
      Metaform foundMetaform = testBuilder.answerer1.metaforms().findMetaform(metaform.getId(), null, null);
      assertNotNull(foundMetaform);
      assertEquals("Simple", foundMetaform.getTitle());
    }
  }

  @Test
  public void testAnswererUpdateForm() throws Exception {
    try (TestBuilder testBuilder = new TestBuilder()) {
      Metaform metaform = testBuilder.metaformAdmin.metaforms().createFromJsonFile("simple");
      testBuilder.answerer1.metaforms().assertUpdateFailStatus(403, metaform);
    }
  }

  @Test
  public void testAnswererDeleteForm() throws Exception {
    try (TestBuilder testBuilder = new TestBuilder()) {
      Metaform metaform = testBuilder.metaformAdmin.metaforms().createFromJsonFile("simple");
      testBuilder.answerer1.metaforms().assertDeleteFailStatus(403, metaform);
    }
  }

  @Test
  public void testAllowAnonymousReply() throws Exception {
    try (TestBuilder testBuilder = new TestBuilder()) {
      Metaform metaform = testBuilder.metaformAdmin.metaforms().createFromJsonFile("simple-allow-anonymous");
      Reply reply = testBuilder.answerer1.replies().createSimpleReply(metaform, "TEST", ReplyMode.REVISION);
      assertNotNull(reply);
      testBuilder.metaformAdmin.replies().delete(metaform.getId(), reply, null);
    }
  }

  @Test
  public void testDisallowAnonymousReply() throws Exception {
    try (TestBuilder testBuilder = new TestBuilder()) {
      Metaform metaform = testBuilder.metaformAdmin.metaforms().createFromJsonFile("simple");
      Reply reply = testBuilder.answerer1.replies().createSimpleReply(metaform, "TEST", ReplyMode.REVISION);
      assertNotNull(reply);
      testBuilder.metaformAdmin.replies().delete(metaform.getId(), reply, null);
    }
  }

}

File: src/test/java/fi/metatavu/metaform/server/test/functional/tests/AnswererTestsIT.kt
package fi.metatavu.metaform.server.test.functional.tests

import fi.metatavu.metaform.api.client.models.Metaform
import fi.metatavu.metaform.api.client.models.Reply
import fi.metatavu.metaform.server.rest.ReplyMode
import fi.metatavu.metaform.server.test.functional.AbstractTest
import fi.metatavu.metaform.server.test.functional.builder.TestBuilder
import fi.metatavu.metaform.server.test.functional.builder.resources.KeycloakResource
import fi.metatavu.metaform.server.test.functional.builder.resources.MysqlResource
import io.quarkus.test.common.QuarkusTestResource
import io.quarkus.test.junit.QuarkusTest
import io.quarkus.test.junit.TestProfile
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

/**
 * Tests for testing answerer user form functionality
 */
@QuarkusTest
@QuarkusTestResource.List(
        QuarkusTestResource(MysqlResource::class),
        QuarkusTestResource(KeycloakResource::class)
)
@TestProfile(GeneralTestProfile::class)
class AnswererTestsIT : AbstractTest() {
//    @Test
//    @Throws(Exception::class)
//    fun testAnswererCreateForm() {
//        TestBuilder().use { testBuilder ->
//            val payload = Metaform()
//            testBuilder.answerer1.metaforms.assertCreateFailStatus(403, payload)
//        }
//    }
//
//    @Test
//    @Throws(Exception::class)
//    fun testAnswererFindForm() {
//        TestBuilder().use { testBuilder ->
//            val metaform = testBuilder.metaformAdmin.metaforms.createFromJsonFile("simple")
//            val foundMetaform: Metaform = testBuilder.answerer1.metaforms.findMetaform(metaform.id!!, null, null)
//            Assertions.assertNotNull(foundMetaform)
//            Assertions.assertEquals("Simple", foundMetaform.title)
//        }
//    }
//
//    @Test
//    @Throws(Exception::class)
//    fun testAnswererUpdateForm() {
//        TestBuilder().use { testBuilder ->
//            val metaform: Metaform = testBuilder.metaformAdmin.metaforms.createFromJsonFile("simple")
//            testBuilder.answerer1.metaforms.assertUpdateFailStatus(403, metaform.id!!, metaform)
//        }
//    }
//
//    @Test
//    @Throws(Exception::class)
//    fun testAnswererDeleteForm() {
//        TestBuilder().use { testBuilder ->
//            val metaform: Metaform = testBuilder.metaformAdmin.metaforms.createFromJsonFile("simple")
//            testBuilder.answerer1.metaforms.assertDeleteFailStatus(403, metaform.id!!)
//        }
//    }
//
//    @Test
//    @Throws(Exception::class)
//    fun testAllowAnonymousReply() {
//        TestBuilder().use { testBuilder ->
//            val metaform: Metaform = testBuilder.metaformAdmin.metaforms.createFromJsonFile("simple-allow-anonymous")
//            val reply: Reply = testBuilder.anon.replies.createSimpleReply(metaform.id!!, "TEST", ReplyMode.REVISION)
//            Assertions.assertNotNull(reply)
//            testBuilder.metaformAdmin.replies.delete(metaform.id, reply.id!!, null)
//        }
//    }
//
//    @Test
//    @Throws(Exception::class)
//    fun testDisallowAnonymousReply() {
//        TestBuilder().use { testBuilder ->
//            val metaform: Metaform = testBuilder.metaformAdmin.metaforms.createFromJsonFile("simple")
//            val reply: Reply = testBuilder.test1.replies.createSimpleReply(metaform.id!!, "TEST", ReplyMode.REVISION)
//            Assertions.assertNotNull(reply)
//            testBuilder.metaformAdmin.replies.delete(metaform.id, reply.id!!, null)
//        }
//    }
}